e3sm_diags/driver/arm_diags_driver.py

```diff
@@ -473,73 +477,174 @@ def run_diag_annual_cycle_aerosol(parameter: ARMDiagsParameter) -> ARMDiagsParam
+    ----------
+    ds : xr.Dataset
+        The dataset.
+    var : str
+        The key of the variable.
+
+    Returns
+    -------
+    Dict[Tuple[str], Callable]
+        A tuple of the derived variable and the function to calculate it.
+    """
+    vars_to_func_dict = DERIVED_VARIABLES[var]
+    vars_in_file = set(ds.keys())
 
+    # e.g,. [('pr',), ('PRECC', 'PRECL')]
+    possible_vars: List[Tuple[str]] = list(vars_to_func_dict.keys())  # type: ignore
 
-def run_diag(parameter: ARMDiagsParameter) -> ARMDiagsParameter:
-    if parameter.diags_set == "annual_cycle":
-        return run_diag_annual_cycle(parameter)
-    elif parameter.diags_set == "diurnal_cycle":
-        return run_diag_diurnal_cycle(parameter)
-    elif parameter.diags_set == "diurnal_cycle_zt":
-        return run_diag_diurnal_cycle_zt(parameter)
-    elif parameter.diags_set == "pdf_daily":
-        return run_diag_pdf_daily(parameter)
-    elif parameter.diags_set == "convection_onset":
-        return run_diag_convection_onset(parameter)
-    elif parameter.diags_set == "aerosol_activation":
-        return run_diag_aerosol_activation(parameter)
-    if parameter.diags_set == "annual_cycle_aerosol":
-        return run_diag_annual_cycle_aerosol(parameter)
-    else:
-        raise Exception("Invalid diags_set={}".format(parameter.diags_set))
+    for list_of_vars in possible_vars:
+        if vars_in_file.issuperset(list_of_vars):
+            return {list_of_vars: vars_to_func_dict[list_of_vars]}
+
+    raise RuntimeError(f"No derived variable function found for the variable {var}")
+
+
+def _get_metrics_dict(
+    test_var: xr.DataArray | np.ndarray, ref_var: xr.DataArray | np.ndarray
+) -> Dict[str, float]:
+    """Calculate various metrics between the test and reference DataArrays.
+
+    Parameters
+    ----------
+    test_var : xr.DataArray | np.ndarray
+        The test variable.
+    ref_var : xr.DataArray | np.ndarray
+        The reference variable.
+
+    Returns
+    -------
+    metrics_dict : Dict[str, float]
+        A dictionary containing the calculated metrics:
+        - 'test_mean': The mean of the test_var.
+        - 'ref_mean': The mean of the ref_var.
+        - 'test_std': The standard deviation of the test_var.
+        - 'ref_std': The standard deviation of the ref_var.
+        - 'rmse': The root mean squared error between the test and ref vars.
+        - 'corr': The correlation coefficient between the test and ref vars.
+    """
+    return {
+        "test_mean": float(test_var.mean()),
+        "ref_mean": float(ref_var.mean()),
+        "test_std": float(test_var.std(ddof=1)),
+        "ref_std": float(ref_var.std(ddof=1)),
+        "rmse": float(_rmse(test_var, ref_var)),
+        "corr": float(np.corrcoef(test_var, ref_var)[0, 1]),
+    }
+
+
+def _rmse(predictions, targets):
+    return np.sqrt(((predictions - targets) ** 2).mean())
+
+
+def _save_metrics_to_json(parameter: ARMDiagsParameter, metrics_dict: Dict[str, float]):
```

Comment:
Re-usable `_save_metrics_to_json` function that removes repeated logic from the diags functions.